doozer/doozerlib/backend/konflux_image_builder.py

```diff
@@ -0,0 +1,365 @@
+        default_revision = f"art-{self._group_name}-assembly-test-dgk-{metadata.distgit_key}"
+
+        component_manifest = self._new_component(
+            component_name,
+            app_name,
+            component_name,
+            dest_image_repo,
+            git_url,
+            default_revision,
+        )
+        component = await self._create_or_patch(dyn_client, component_manifest)
+        self._logger.info(f"Created component: {component['metadata']['name']}")
+
+        # Create a PipelineRun
+        pipelineruns_api = await self._get_pipelinerun_api(dyn_client)
+        pipelinerun_manifest = self._new_pipelinerun(
+            f"doozer-build-{component_name}-",
+            app_name,
+            component_name,
+            git_url,
+            git_commit,
+            git_branch,
+            f"{dest_image_repo}:{dest_image_tag}"
+        )
+
+        if self._dry_run:
+            pipelinerun_manifest = resource.ResourceInstance(dyn_client, pipelinerun_manifest)
+            pipelinerun_manifest.metadata.name = f"doozer-build-{component_name}-dry-run"
+            self._logger.warning(f"[DRY RUN] Would have created PipelineRun: {pipelinerun_manifest.metadata.name}")
+            return pipelinerun_manifest
+
+        pipelinerun = await exectools.to_thread(
+            pipelineruns_api.create,
+            namespace=self._konflux_namespace,
+            body=pipelinerun_manifest,
+            async_req=True,
+        )
+        pipelinerun = cast(resource.ResourceInstance, pipelinerun)
+
+        pipelinerun_name = pipelinerun['metadata']['name']
+        self._logger.info(f"Created PipelineRun: {pipelinerun_name}")
+        return pipelinerun
+
+    @staticmethod
+    def _new_application(name: str, display_name: str) -> dict:
+        obj = {
+            "apiVersion": "appstudio.redhat.com/v1alpha1",
+            "kind": "Application",
+            "metadata": {
+                "name": name
+            },
+            "spec": {
+                "displayName": display_name,
+                "appModelRepository": {"url": ""},
+                "gitOpsRepository": {"url": ""},
+            }
+        }
+        return obj
+
+    @staticmethod
+    async def _get_pipelinerun_api(dyn_client: DynamicClient):
+        return await exectools.to_thread(
+            dyn_client.resources.get,
+            api_version="tekton.dev/v1",
+            kind="PipelineRun",
+        )
+
+    @staticmethod
+    def _new_component(name: str, application: str, component_name: str,
+                       image_repo: Optional[str], source_url: Optional[str], revision: Optional[str]) -> dict:
+        obj = {
+            "apiVersion": "appstudio.redhat.com/v1alpha1",
+            "kind": "Component",
+            "metadata": {
+                "name": name,
+                "annotations": {
+                    "build.appstudio.openshift.io/pipeline": '{"name":"docker-build","bundle":"latest"}',
+                }
```

Comment:
```
mintmaker.appstudio.redhat.com/disabled=true
```
Need to add this annotation in, just to be safe. ref [thread](https://redhat-internal.slack.com/archives/GDBRP5YJH/p1724260588819889?thread_ts=1723123622.362349&cid=GDBRP5YJH)